internal/codegen/golang/imports.go

```diff
@@ -462,15 +460,9 @@ func (i *importer) batchImports() fileImports {
 	return sortedImports(std, pkg)
 }
 
-func trimSliceAndPointerPrefix(v string) string {
-	v = strings.TrimPrefix(v, "[]")
-	v = strings.TrimPrefix(v, "*")
-	return v
-}
-
 func hasPrefixIgnoringSliceAndPointerPrefix(s, prefix string) bool {
-	trimmedS := trimSliceAndPointerPrefix(s)
-	trimmedPrefix := trimSliceAndPointerPrefix(prefix)
+	trimmedS := strings.TrimLeft(s, "*[]")
+	trimmedPrefix := strings.TrimLeft(prefix, "*[]")
```

Comment:
Do we validate these prefixes anywhere? With the change to `TrimLeft`, I think a string like this `****[[]]]database/foo` will work?

Reply:
We do not validate these anywhere as far as I can see. You are right about this change trimming more than is perhaps desirable, so I'll revert.